sys/riscv/riscv/locore.S

```diff
@@ -415,12 +461,6 @@ va:
 	addi	s0, s0, 8
 	bltu	s0, s1, 1b
 
-#ifdef SMP
-	/* Store boot hart id. */
-	la	t0, boot_hart
-	sw	a0, 0(t0)
-#endif
-
 	/* Save DTB physical pointer in s1. */
 	mv	s1, a1
 
```

Comment:
Need to save off `a0` somewhere since it gets clobbered below.  Perhaps save it in `s2` and then store `s2` to RISCV_BOOTPARAMS_MODULEP(sp) below.

Reply:
Ah yeah, although I guess currently we don't use this but start at `_alt_start`?